components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java

```diff
@@ -146,8 +146,9 @@ public long getAPISubscriptionCountByAPI(APIIdentifier identifier)
      *
      * @param api API
      * @throws org.wso2.carbon.apimgt.api.APIManagementException if failed to update API
+     * @return  failed environments during gateway operation
      */
-    public void updateAPI(API api) throws APIManagementException;
+    public Map<String, List<String>> updateAPI(API api) throws APIManagementException;
```

Comment:
Can you point to the PR where we changed this logic by getting the failed list of Envs through an exception?